crates/lang/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+// ------ Lang ------
+
+use std::borrow::Cow;
+use std::fmt;
+
+#[derive(Clone)]
+pub enum Lang {
+    Czech,
+    English,
+    French,
+    Norwegian,
+    Spanish,
+    Swedish,
+    Custom(Cow<'static, str>),
+}
+
+impl Lang {
+    pub fn as_str(&self) -> &str {
+        match self {
+            Self::Czech => "cs",
+            Self::English => "en",
+            Self::French => "fr",
+            Self::Norwegian => "no",
+            Self::Spanish => "es",
+            Self::Swedish => "sv",
+            Self::Custom(str) => str,
```

Comment:
I understand your motivation why you've decided to rename `lang` to `str` (to eliminate all `lang` occurrences in the  body during the refactor to `Self`s) but I think `str` isn't a very good choice because it would be confusing for devs (they would think the type is `str` at first glance) and "dumb" code highlighters (like this one on GitHub) highlight it because it's a Rust built-in type. (Sorry for a too long sentence 😅.)  I'm quite surprised it's compilable at all I would guess `str` is a keyword you can't use as an identifier.

So perhaps let's return to the `lang` name or maybe `custom` or something else would be better.

Reply:
I think I was just trying stuff out and then forgot to change it back to a good variable name oops 😅